diff -Naur live.old/liveMedia/include/RTPSource.hh live/liveMedia/include/RTPSource.hh
--- live.old/liveMedia/include/RTPSource.hh	2023-01-19 16:01:59.000000000 +0100
+++ live/liveMedia/include/RTPSource.hh	2024-02-26 16:42:21.733370235 +0100
@@ -86,7 +86,7 @@
   // RTP sequence numbers and timestamps are usually not useful to receivers.
   // (Our implementation of RTP reception already does all needed handling of RTP sequence numbers and timestamps.)
   u_int16_t curPacketRTPSeqNum() const { return fCurPacketRTPSeqNum; }
-private: friend class MediaSubsession; // "MediaSubsession" is the only outside class that ever needs to see RTP timestamps!
+//private: friend class MediaSubsession; // "MediaSubsession" is the only outside class that ever needs to see RTP timestamps!
   u_int32_t curPacketRTPTimestamp() const { return fCurPacketRTPTimestamp; }
 
 protected:
