--- mqttv4.ori/src/files.c	2022-06-10 12:47:22.775769275 +0200
+++ mqttv4/src/files.c	2022-06-10 12:46:58.000000000 +0200
@@ -89,8 +89,8 @@
     sprintf(output + strlen(output), "\"%s%c%c%c:%c%c\",\n\"files\":[ ", s8601date,
             s8601date_z[0], s8601date_z[1], s8601date_z[2], s8601date_z[3], s8601date_z[4]);
 
-    // Use localtime to read record directories
-    timeinfo=localtime(&endTime);
+    // Use gmt time to read record directories
+    timeinfo=gmtime(&endTime);
     sprintf(sDir, "%dY%02dM%02dD%02dH", timeinfo->tm_year + 1900,
         timeinfo->tm_mon + 1, timeinfo->tm_mday, timeinfo->tm_hour);
     sprintf(sFile, "%02dM", timeinfo->tm_min);
@@ -102,9 +102,9 @@
         num++;
         if (num >= limit) break;
 
-        rawtime=timelocal(timeinfo);
+        rawtime=timegm(timeinfo);
         rawtime-=60;
-        timeinfo=localtime(&rawtime);
+        timeinfo=gmtime(&rawtime);
 
         if (rawtime < (startTime - 60)) break;
 
