diff -Naur tinyalsa-1.0.0.ori/include/tinyalsa/pcm.h tinyalsa-1.0.0/include/tinyalsa/pcm.h
--- tinyalsa-1.0.0.ori/include/tinyalsa/pcm.h	2016-10-02 04:37:39.000000000 +0200
+++ tinyalsa-1.0.0/include/tinyalsa/pcm.h	2021-03-19 14:42:12.901718483 +0100
@@ -218,6 +218,7 @@
                      unsigned int flags, struct pcm_config *config);
 int pcm_close(struct pcm *pcm);
 int pcm_is_ready(struct pcm *pcm);
+int pcm_is_xrun(struct pcm *pcm);
 
 struct pcm_params *pcm_params_get(unsigned int card, unsigned int device,
                                   unsigned int flags);
@@ -262,6 +263,9 @@
 
 long pcm_get_delay(struct pcm *pcm);
 
+int pcm_create_snoop_fifo(struct pcm *pcm);
+int pcm_write_snoop_fifo(struct pcm *pcm, void *data, unsigned int bytes);
+int pcm_write_snoop_fifo_wavhdr(struct pcm *pcm);
 #if defined(__cplusplus)
 }  /* extern "C" */
 #endif
diff -Naur tinyalsa-1.0.0.ori/include/tinyalsa/wave.h tinyalsa-1.0.0/include/tinyalsa/wave.h
--- tinyalsa-1.0.0.ori/include/tinyalsa/wave.h	1970-01-01 01:00:00.000000000 +0100
+++ tinyalsa-1.0.0/include/tinyalsa/wave.h	2021-03-19 14:42:12.901718483 +0100
@@ -0,0 +1,31 @@
+#ifndef WAVE_H
+#define WAVE_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#define ID_RIFF 0x46464952
+#define ID_WAVE 0x45564157
+#define ID_FMT  0x20746d66
+#define ID_DATA 0x61746164
+
+#define FORMAT_PCM 1
+
+struct wav_header {
+    uint32_t riff_id;
+    uint32_t riff_sz;
+    uint32_t riff_fmt;
+    uint32_t fmt_id;
+    uint32_t fmt_sz;
+    uint16_t audio_format;
+    uint16_t num_channels;
+    uint32_t sample_rate;
+    uint32_t byte_rate;
+    uint16_t block_align;
+    uint16_t bits_per_sample;
+    uint32_t data_id;
+    uint32_t data_sz;
+};
+
+#endif
diff -Naur tinyalsa-1.0.0.ori/Makefile tinyalsa-1.0.0/Makefile
--- tinyalsa-1.0.0.ori/Makefile	2016-10-02 04:37:39.000000000 +0200
+++ tinyalsa-1.0.0/Makefile	2021-03-19 14:42:12.901718483 +0100
@@ -10,13 +10,13 @@
 all:
 	$(MAKE) -C src
 	$(MAKE) -C utils
-	$(MAKE) -C doxygen
+#	$(MAKE) -C doxygen
 
 .PHONY: clean
 clean:
 	$(MAKE) -C src clean
 	$(MAKE) -C utils clean
-	$(MAKE) -C doxygen clean
+#	$(MAKE) -C doxygen clean
 
 .PHONY: install
 install: $(LIB) $(SHLIB)
@@ -24,5 +24,5 @@
 	cp -Ru $(INCDIR)/tinyalsa $(DESTDIR)$(PREFIX)/include/
 	$(MAKE) -C src install
 	$(MAKE) -C utils install
-	$(MAKE) -C doxygen install
+#	$(MAKE) -C doxygen install
 
diff -Naur tinyalsa-1.0.0.ori/src/pcm.c tinyalsa-1.0.0/src/pcm.c
--- tinyalsa-1.0.0.ori/src/pcm.c	2016-10-02 04:37:39.000000000 +0200
+++ tinyalsa-1.0.0/src/pcm.c	2021-03-19 14:46:00.741456522 +0100
@@ -26,6 +26,8 @@
 ** DAMAGE.
 */
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <fcntl.h>
@@ -34,12 +36,16 @@
 #include <errno.h>
 #include <unistd.h>
 #include <poll.h>
+#include <signal.h>
 
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 #include <sys/time.h>
 #include <time.h>
 #include <limits.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <pthread.h>
 
 #include <linux/ioctl.h>
 #define __force
@@ -48,10 +54,18 @@
 #include <sound/asound.h>
 
 #include <tinyalsa/asoundlib.h>
+#include <tinyalsa/wave.h>
 
 #define PARAM_MAX SNDRV_PCM_HW_PARAM_LAST_INTERVAL
 #define SNDRV_PCM_HW_PARAMS_NO_PERIOD_WAKEUP (1<<2)
 
+#define CONVERT_TO_MONO
+//#define CONVERT_TO_8KHZ
+#ifdef CONVERT_TO_8KHZ
+#define CONVERT_TO_MONO
+#endif
+//#define START_WAV_HEADER
+
 static inline int param_is_mask(int p)
 {
     return (p >= SNDRV_PCM_HW_PARAM_FIRST_MASK) &&
@@ -186,6 +200,11 @@
     unsigned int subdevice;
 };
 
+/* Handle for the pipe that will be used to clone the audio frames.
+ * Cannot be inside the pcm struct since it crashes to rmm application
+ */
+int snoop_fd = 0;
+
 /** Gets the buffer size of the PCM.
  * @param pcm A PCM handle.
  * @return The buffer size of the PCM.
@@ -556,12 +575,15 @@
             pcm->prepared = 0;
             pcm->running = 0;
             if (errno == EPIPE) {
-                    /* we failed to make our window -- try to restart */
+                /* we failed to make our window -- try to restart */
                 pcm->underruns++;
                 continue;
             }
             return oops(pcm, errno, "cannot read stream data");
         }
+
+        /* Write to fifo, ignore the result */
+        pcm_write_snoop_fifo(pcm, data, count);
         return 0;
     }
 }
@@ -764,6 +786,9 @@
     pcm->running = 0;
     pcm->buffer_size = 0;
     pcm->fd = -1;
+
+    if (snoop_fd > 0)
+        close(snoop_fd);
     free(pcm);
     return 0;
 }
@@ -903,7 +928,7 @@
     pcm->boundary = sparams.boundary = pcm->buffer_size;
 
     while (pcm->boundary * 2 <= INT_MAX - pcm->buffer_size)
-		pcm->boundary *= 2;
+                pcm->boundary *= 2;
 
     if (ioctl(pcm->fd, SNDRV_PCM_IOCTL_SW_PARAMS, &sparams)) {
         oops(pcm, errno, "cannot set sw params");
@@ -928,6 +953,16 @@
 #endif
 
     pcm->underruns = 0;
+
+    if (pcm_create_snoop_fifo(pcm) != 0)
+    {
+            printf("Error creating fifo\n");  
+            oops(pcm, errno, "cannot create snoop fifo");
+            goto fail;
+    }
+    /* Hopefully nobody uses this one */
+    signal(SIGPIPE, SIG_IGN);
+
     return pcm;
 
 fail:
@@ -981,7 +1016,7 @@
         return prepare_error;
 
     if (pcm->flags & PCM_MMAP)
-	    pcm_sync_ptr(pcm, 0);
+            pcm_sync_ptr(pcm, 0);
 
     if (ioctl(pcm->fd, SNDRV_PCM_IOCTL_START) < 0)
         return oops(pcm, errno, "cannot start channel");
@@ -1162,7 +1197,7 @@
         }
 
         /* start the audio if we reach the threshold */
-	    if (!pcm->running &&
+            if (!pcm->running &&
             (pcm->buffer_size - avail) >= pcm->config.start_threshold) {
             if (pcm_start(pcm) < 0) {
                fprintf(stderr, "start error: hw 0x%x app 0x%x avail 0x%x\n",
@@ -1244,3 +1279,145 @@
     return pcm->pcm_delay;
 }
 
+int pcm_is_xrun(struct pcm *pcm)
+{
+    if (pcm->underruns > 0)
+    {
+        pcm->underruns = 0;
+        return 1;
+    }
+
+    return 0;
+}
+
+void* fifo_open_thread(void *data)
+{
+    int fd;
+    char *fifo_name = data;
+
+    fd = open(fifo_name, O_RDONLY);
+    usleep(250);
+    close(fd);
+
+    return NULL;
+}
+
+int pcm_create_snoop_fifo(struct pcm *pcm)
+{
+    char *fifo = "/tmp/audio_fifo";
+    pthread_t open_thread;
+
+    if (mkfifo(fifo, 0666) < 0) {
+        if (errno != EEXIST) {
+            oops(pcm, errno, "cannot create snoop fifo");
+            return -1;
+        }
+    }
+
+    if(pthread_create(&open_thread, NULL, fifo_open_thread, fifo)) {
+        oops(pcm, errno, "Error creating thread\n");
+        return -1;
+    }
+
+    snoop_fd = open(fifo, O_WRONLY);
+    if (snoop_fd <= 0) {
+        oops(pcm, errno, "cannot open snoop fifo");
+        return -1;
+    }
+
+    if(pthread_join(open_thread, NULL)) {
+        oops(pcm, errno, "Failed to join unblocking thread");
+        return -1;
+    }
+
+    if (fcntl(snoop_fd, F_SETFL, O_NONBLOCK) != 0) {
+        oops(pcm, errno, "cannot set non-block to snoop fifo");
+        return -1;
+    };
+
+    if (fcntl(snoop_fd, F_SETPIPE_SZ, 8192) != 8192) {
+        oops(pcm, errno, "cannot set non-block to snoop fifo");
+        return -1;
+    };
+#ifdef START_WAV_HEADER
+    if (pcm_write_snoop_fifo_wavhdr(pcm) != 0) {
+        oops(pcm, errno, "failed to write wave header");
+        return -1;
+    }
+#endif
+    return 0;
+}
+
+int pcm_write_snoop_fifo(struct pcm *pcm, void *data, unsigned int bytes)
+{
+#ifdef CONVERT_TO_MONO
+    int16_t mono_buf[2048]; /* rmm reads 512 frames per call, 2k to be sure */
+    uint32_t *pcm_ptr = (uint32_t*)data;
+    uint32_t i;
+#endif
+
+    if (snoop_fd <= 0) {
+        oops(pcm, errno, "snoop fifo is not openend");
+        return -1;
+    }
+
+#ifdef CONVERT_TO_MONO
+    for (i = 0; i < pcm_bytes_to_frames(pcm, bytes); i++) {
+        mono_buf[i] = *pcm_ptr & 0xffff;
+        //mono_buf[i] = cnt++;
+        pcm_ptr++;
+#ifdef CONVERT_TO_8KHZ
+        pcm_ptr++;
+#endif
+    }
+
+#ifdef CONVERT_TO_8KHZ
+    if (write(snoop_fd, mono_buf, bytes / 4) < 0) {
+#else
+    if (write(snoop_fd, mono_buf, bytes / 2) < 0) {
+#endif
+#else
+    if (write(snoop_fd, data, bytes) < 0) {
+#endif
+        if (errno == EAGAIN || errno == EWOULDBLOCK) {
+            oops(pcm, errno, "Fifo full!\n");
+            return -1;
+        }
+    }
+    return 0;
+}
+
+int pcm_write_snoop_fifo_wavhdr(struct pcm *pcm)
+{
+    struct wav_header header;
+
+    header.riff_id = ID_RIFF;
+    header.riff_sz = 0;
+    header.riff_fmt = ID_WAVE;
+    header.fmt_id = ID_FMT;
+    header.fmt_sz = 16;
+    header.audio_format = FORMAT_PCM;
+#ifdef CONVERT_TO_MONO
+    header.num_channels = 1;
+#else
+    header.num_channels = 2;
+#endif
+#ifdef CONVERT_TO_8KHZ
+    header.sample_rate = 8000;
+#else
+    header.sample_rate = 16000;
+#endif
+    header.bits_per_sample = pcm_format_to_bits(PCM_FORMAT_S16_LE);
+    header.byte_rate = (header.bits_per_sample / 8) * header.num_channels * header.sample_rate;
+    header.block_align = header.num_channels * (header.bits_per_sample / 8);
+    header.data_id = ID_DATA;
+    unsigned int frames = 0;
+    header.data_sz = frames * header.block_align;
+    header.riff_sz = header.data_sz + sizeof(header) - 8;
+
+    if (write(snoop_fd, &header, sizeof(header)) < 0) {
+        oops(pcm, errno, "cannot write wav header\n");
+        return -1;
+    }
+    return 0;
+}
