diff -Naur busybox-1.36.1.ori/networking/httpd.c busybox-1.36.1/networking/httpd.c
--- busybox-1.36.1.ori/networking/httpd.c	2023-01-03 15:14:43.000000000 +0100
+++ busybox-1.36.1/networking/httpd.c	2024-01-08 16:45:49.504118695 +0100
@@ -2182,7 +2182,7 @@
 	struct stat sb;
 	char *urlcopy;
 	char *urlp;
-	char *tptr;
+	char *tptr, *gztptr;
 #if ENABLE_FEATURE_HTTPD_ACL_IP
 	unsigned remote_ip;
 #endif
@@ -2418,7 +2418,32 @@
 			g_query = xstrdup(g_query); /* ok for NULL too */
 		strcpy(urlp, index_page);
 	}
+	gztptr = xasprintf("%s.gz", tptr);
 	if (stat(tptr, &sb) == 0) {
+		free(gztptr);
+		/* If URL is a directory with no slash, set up
+		 * "HTTP/1.1 302 Found" "Location: /dir/" reply */
+		if (urlp[-1] != '/' && S_ISDIR(sb.st_mode)) {
+			found_moved_temporarily = urlcopy;
+		} else {
+#if ENABLE_FEATURE_HTTPD_CONFIG_WITH_SCRIPT_INTERPR
+			char *suffix = strrchr(tptr, '.');
+			if (suffix) {
+				Htaccess *cur;
+				for (cur = script_i; cur; cur = cur->next) {
+					if (strcmp(cur->before_colon + 1, suffix) == 0) {
+						cgi_type = CGI_INTERPRETER;
+						break;
+					}
+				}
+			}
+#endif
+			file_size = sb.st_size;
+			last_mod = sb.st_mtime;
+		}
+	}
+	else if (stat(gztptr, &sb) == 0) {
+		free(gztptr);
 		/* If URL is a directory with no slash, set up
 		 * "HTTP/1.1 302 Found" "Location: /dir/" reply */
 		if (urlp[-1] != '/' && S_ISDIR(sb.st_mode)) {
@@ -2442,6 +2467,7 @@
 	}
 #if ENABLE_FEATURE_HTTPD_CGI
 	else if (urlp[-1] == '/') {
+		free(gztptr);
 		/* It's a dir URL and there is no index.html */
 		/* Is there cgi-bin/index.cgi? */
 		if (access("/cgi-bin/index.cgi"+1, X_OK) != 0)
