diff -Naur onvif_simple_server.ori/onvif_notify_server.c onvif_simple_server/onvif_notify_server.c
--- onvif_simple_server.ori/onvif_notify_server.c	2023-11-07 21:36:47.640210252 +0100
+++ onvif_simple_server/onvif_notify_server.c	2023-11-07 21:38:59.448510264 +0100
@@ -39,10 +39,10 @@
 #include "log.h"
 #include "onvif_simple_server.h"
 
-#define DEFAULT_CONF_FILE "/etc/onvif_simple_server.conf"
-#define DEFAULT_LOG_FILE "/var/log/onvif_notify_server.log"
+#define DEFAULT_CONF_FILE "/tmp/onvif_simple_server.conf"
+#define DEFAULT_LOG_FILE "/tmp/onvif_notify_server.log"
 #define DEFAULT_PID_FILE "/var/run/onvif_notify_server.pid"
-#define TEMPLATE_DIR "/etc/onvif_notify_server"
+#define TEMPLATE_DIR "/home/yi-hack/etc/onvif_notify_server"
 #define INOTIFY_DIR "/tmp/onvif_notify_server"
 
 #define BD_NO_CHDIR          01
diff -Naur onvif_simple_server.ori/onvif_simple_server.c onvif_simple_server/onvif_simple_server.c
--- onvif_simple_server.ori/onvif_simple_server.c	2023-11-07 21:36:47.640210252 +0100
+++ onvif_simple_server/onvif_simple_server.c	2023-11-07 21:39:20.952557004 +0100
@@ -36,8 +36,8 @@
 #include "utils.h"
 #include "log.h"
 
-#define DEFAULT_CONF_FILE "/etc/onvif_simple_server.conf"
-#define DEFAULT_LOG_FILE "/var/log/onvif_simple_server.log"
+#define DEFAULT_CONF_FILE "/tmp/onvif_simple_server.conf"
+#define DEFAULT_LOG_FILE "/tmp/onvif_simple_server.log"
 #define DEBUG_FILE "/tmp/onvif_simple_server.debug"
 
 service_context_t service_ctx;
diff -Naur onvif_simple_server.ori/wsd_simple_server.c onvif_simple_server/wsd_simple_server.c
--- onvif_simple_server.ori/wsd_simple_server.c	2023-11-07 21:36:47.640210252 +0100
+++ onvif_simple_server/wsd_simple_server.c	2023-11-07 21:39:36.432590305 +0100
@@ -37,8 +37,8 @@
 #define PORT 3702
 #define TYPE "NetworkVideoTransmitter"
 
-#define DEFAULT_LOG_FILE "/var/log/wsd_simple_server.log"
-#define TEMPLATE_DIR "/etc/wsd_simple_server"
+#define DEFAULT_LOG_FILE "/tmp/wsd_simple_server.log"
+#define TEMPLATE_DIR "/home/yi-hack/etc/wsd_simple_server"
 
 #define RECV_BUFFER_LEN 4096
 
