diff -Naur onvif_simple_server.ori/onvif_simple_server.c onvif_simple_server/onvif_simple_server.c
--- onvif_simple_server.ori/onvif_simple_server.c	2023-10-01 17:33:29.280693498 +0200
+++ onvif_simple_server/onvif_simple_server.c	2023-10-01 17:31:49.728491342 +0200
@@ -32,8 +32,8 @@
 #include "utils.h"
 #include "log.h"
 
-#define DEFAULT_CONF_FILE "/etc/onvif_simple_server.conf"
-#define DEFAULT_LOG_FILE "/var/log/onvif_simple_server.log"
+#define DEFAULT_CONF_FILE "/tmp/onvif_simple_server.conf"
+#define DEFAULT_LOG_FILE "/tmp/onvif_simple_server.log"
 #define DEBUG_FILE "/tmp/onvif_simple_server.debug"
 
 service_context_t service_ctx;
diff -Naur onvif_simple_server.ori/wsd_simple_server.c onvif_simple_server/wsd_simple_server.c
--- onvif_simple_server.ori/wsd_simple_server.c	2023-10-01 17:33:47.192729869 +0200
+++ onvif_simple_server/wsd_simple_server.c	2023-10-01 17:32:33.216579645 +0200
@@ -36,8 +36,8 @@
 #define PORT 3702
 #define TYPE "NetworkVideoTransmitter"
 
-#define DEFAULT_LOG_FILE "/var/log/wsd_simple_server.log"
-#define TEMPLATE_DIR "/etc/wsd_simple_server"
+#define DEFAULT_LOG_FILE "/tmp/wsd_simple_server.log"
+#define TEMPLATE_DIR "/home/yi-hack/etc/wsd_simple_server"
 
 #define UUID_LEN 36
 #define RECV_BUFFER_LEN 4096
