diff -Naur onvif_simple_server.ori/onvif_notify_server.c onvif_simple_server/onvif_notify_server.c
--- onvif_simple_server.ori/onvif_notify_server.c	2025-05-20 16:04:41.566106647 +0200
+++ onvif_simple_server/onvif_notify_server.c	2025-05-19 21:58:06.946512597 +0200
@@ -39,11 +39,11 @@
 #include "log.h"
 #include "onvif_simple_server.h"
 
-#define DEFAULT_CONF_FILE "/etc/onvif_simple_server.conf"
-#define DEFAULT_JSON_CONF_FILE "/etc/onvif_simple_server.json"
-#define DEFAULT_LOG_FILE "/var/log/onvif_notify_server.log"
+#define DEFAULT_CONF_FILE "/tmp/onvif_simple_server.conf"
+#define DEFAULT_JSON_CONF_FILE "/tmp/onvif_simple_server.json"
+#define DEFAULT_LOG_FILE "/tmp/onvif_notify_server.log"
 #define DEFAULT_PID_FILE "/var/run/onvif_notify_server.pid"
-#define TEMPLATE_DIR "/etc/onvif_notify_server"
+#define TEMPLATE_DIR "/home/yi-hack/etc/onvif_notify_server"
 #define INOTIFY_DIR "/tmp/onvif_notify_server"
 
 #define ALARM_OFF 0
diff -Naur onvif_simple_server.ori/onvif_simple_server.c onvif_simple_server/onvif_simple_server.c
--- onvif_simple_server.ori/onvif_simple_server.c	2025-05-20 16:05:41.953491940 +0200
+++ onvif_simple_server/onvif_simple_server.c	2025-05-19 22:11:39.609081482 +0200
@@ -38,9 +38,9 @@
 #include "utils.h"
 #include "log.h"
 
-#define DEFAULT_CONF_FILE "/etc/onvif_simple_server.conf"
-#define DEFAULT_JSON_CONF_FILE "/etc/onvif_simple_server.json"
-#define DEFAULT_LOG_FILE "/var/log/onvif_simple_server.log"
+#define DEFAULT_CONF_FILE "/tmp/onvif_simple_server.conf"
+#define DEFAULT_JSON_CONF_FILE "/tmp/onvif_simple_server.json"
+#define DEFAULT_LOG_FILE "/tmp/onvif_simple_server.log"
 #define DEBUG_FILE "/tmp/onvif_simple_server.debug"
 
 #define ROTATION_LOG_LENGTH 3
diff -Naur onvif_simple_server.ori/wsd_simple_server.c onvif_simple_server/wsd_simple_server.c
--- onvif_simple_server.ori/wsd_simple_server.c	2025-05-20 16:06:31.857058818 +0200
+++ onvif_simple_server/wsd_simple_server.c	2025-05-19 20:17:54.965288550 +0200
@@ -38,8 +38,8 @@
 #define PORT 3702
 #define TYPE "NetworkVideoTransmitter"
 
-#define DEFAULT_LOG_FILE "/var/log/wsd_simple_server.log"
-#define TEMPLATE_DIR "/etc/wsd_simple_server"
+#define DEFAULT_LOG_FILE "/tmp/wsd_simple_server.log"
+#define TEMPLATE_DIR "/home/yi-hack/etc/wsd_simple_server"
 
 #define RECV_BUFFER_LEN 4096
 
